SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -1684,33 +1684,73 @@ USpatialActorChannel* ActorSystem::TryRestoreActorChannelForStablyNamedActor(AAc
 
 void ActorSystem::InvokeRepNotifies()
 {
-	for (FObjectRepNotifies& ObjectRepNotifies : RepNotifiesToSend)
+	// Call repnotifies on actors before subobjects
+	TArray<FObjectRepNotifies> Actors;
+	TArray<FObjectRepNotifies> Subobjects;
+	for (auto ObjectToRepNotifiesIt = RepNotifiesToSend.CreateIterator(); ObjectToRepNotifiesIt; ++ObjectToRepNotifiesIt)
 	{
-		UObject* Object = ObjectRepNotifies.Object.Get();
-		if (!IsValid(Object))
+		FObjectRepNotifies& ObjectRepNotifies = ObjectToRepNotifiesIt->Value;
+		if (ObjectRepNotifies.RepNotifies.Num() == 0)
 		{
 			continue;
 		}
-		const Worker_EntityId EntityId = NetDriver->PackageMap->GetEntityIdFromObject(Object);
-		if (EntityId == SpatialConstants::INVALID_ENTITY_ID)
+
+		UObject* Object = ObjectToRepNotifiesIt->Key.Get();
+		ObjectRepNotifies.Object = Object;
+
+		if (Cast<AActor>(Object))
 		{
-			// TODO UNR-5785 - Once fixed, change log level to warning
-			UE_LOG(LogActorSystem, Log, TEXT("Failed to invoke rep notifies for an object as its entity id was invalid. Object: %s"),
-				   *Object->GetName());
-			continue;
+			Actors.Add(MoveTemp(ObjectRepNotifies));
```

Comment:
Instead of sorting our RepNotifiesToSend into two buckets here, we could have `ActorRepNotifiesToSend` and `SubobjectRepNotifiesToSend`, but adds some complexity to parts of code unrelated to repnotifies. Is it worth?

Reply:
If one or more local arrays is the preferred solution, I would reserve them enough space (based on size of repnotifiestosend) to avoid allocations when adding in loop below, even if it means allocating twice what was actually needed (in the case of using two local arrays)

Reply:
Added reserves in, good idea

Reply:
I assume we are not talking about 1000's of entries?

Reply:
It depends on the game - any object receiving a property update within a given ActorSystem::Advance will make its way here, so depends how many objects are receiving an update in a given tick

Reply:
I haven't seen enough performance profiles in general to guess what perf impacts of different things are, so will leave the decision on this up to you

Reply:
Technically, since actors and subobjects appear to not be sorted here, you only need a container for subobjects, actors can call TryInvokeRepNotifiesForObject immediately? If so, at that points its a profiling decision whether to have that container at all, or just run through RepNotifiesToSend twice... I'd stick with the single container for now personally.

Reply:
whoops, probably should have caught that. changed to a single subobject tarray